sw/device/lib/dif/dif_pinmux.c

```diff
@@ -32,9 +200,304 @@ dif_result_t dif_pinmux_output_select(const dif_pinmux_t *pinmux,
       outsel >= (2 + PINMUX_PARAM_N_MIO_PERIPH_OUT)) {
     return kDifBadArg;
   }
-  uint32_t reg_offset = PINMUX_MIO_OUTSEL_0_REG_OFFSET + (mio_pad_output << 2);
+  bool is_locked;
+  dif_result_t result = dif_pinmux_is_locked(
+      pinmux, mio_pad_output, kDifPinmuxLockTargetOutsel, &is_locked);
+  if (result != kDifOk) {
+    return result;
+  }
+  if (is_locked) {
+    return kDifLocked;
+  }
+  ptrdiff_t reg_offset = PINMUX_MIO_OUTSEL_0_REG_OFFSET + (mio_pad_output << 2);
   uint32_t reg_value =
       bitfield_field32_write(0, PINMUX_MIO_OUTSEL_0_OUT_0_FIELD, outsel);
   mmio_region_write32(pinmux->base_addr, reg_offset, reg_value);
   return kDifOk;
 }
+
+dif_result_t dif_pinmux_pad_write_attrs(const dif_pinmux_t *pinmux,
+                                        dif_pinmux_index_t pad,
+                                        dif_pinmux_pad_kind_t type,
+                                        dif_pinmux_pad_attr_mask_t attrs_mask,
+                                        dif_pinmux_pad_attr_t attrs_in,
+                                        dif_pinmux_pad_attr_t *attrs_out) {
+  if (pinmux == NULL || attrs_out == NULL) {
+    return kDifBadArg;
+  }
+  // TODO: Implement
```

Comment:
In a followup, right?

Reply:
Aye. I need to follow up with @msfschaffner about defining the pad attributes in the hjson directly (so it can propagate to reggen) instead of in a comment.

Reply:
Ack. Let me know what information you need and what format would be preferrable.

Reply:
Or since this PR has been around long enough, we'll throw it in!